hapi-fhir-storage-cr/src/main/java/ca/uhn/fhir/cr/r4/measure/CareGapsProvider.java

```
@@ -0,0 +1,143 @@
+	@Operation(name = "$care-gaps", idempotent = true, type = Measure.class)
+	public Parameters careGapsReport(
+		RequestDetails theRequestDetails,
+		@OperationParam(name = "periodStart", typeName = "date") IPrimitiveType<Date> periodStart,
+		@OperationParam(name = "periodEnd", typeName = "date") IPrimitiveType<Date> periodEnd,
+		@OperationParam(name = "topic") List<String> topic,
+		@OperationParam(name = "subject") String subject,
+		@OperationParam(name = "practitioner") String practitioner,
+		@OperationParam(name = "organization") String organization,
+		@OperationParam(name = "status") List<String> status,
+		@OperationParam(name = "measureId") List<String> measureId,
+		@OperationParam(name = "measureIdentifier") List<String> measureIdentifier,
+		@OperationParam(name = "measureUrl") List<CanonicalType> measureUrl,
+		@OperationParam(name = "program") List<String> program) {
+
+		try {
+			validateParameters(theRequestDetails);
+		} catch (Exception e) {
+			return parameters(part("Invalid parameters", generateIssue("error", e.getMessage())));
+		}
+		return myCareGapsServiceFunction
+			.apply(theRequestDetails)
+					.getCareGapsReport(
+						periodStart,
+						periodEnd,
+						topic,
+						subject,
+						practitioner,
+						organization,
+						status,
+						measureId,
+						measureIdentifier,
+						measureUrl,
+						program
+					);
+	}
+
+	OperationOutcome generateIssue(String severity, String issue) {
```

There is likely a utility class for generating OperationOutcomes.

---

Also, this should throw a meaningful exception, it shouldn't return a successful response in the form of a Parameters resource with an "invalid parameters" part.

---

The HAPI REST processing layer will take care of converting that to an OperationOutcome.

https://build.fhir.org/operations.html#response